Add labels for keeping track of the number of children that belong to a supernode

mozartfish opened the pull request:
https://user-images.githubusercontent.com/24800816/114766208-f92b3100-9d1a-11eb-81aa-9fbc152ba34b.mp4



Child Label 
![image](https://user-images.githubusercontent.com/24800816/114446049-a4a27d00-9b85-11eb-8875-298703880d71.png)

JackWilb commented:
Instead of making another variable on the nodes, couldn't you just use `CHILDREN.length`?

JackWilb commented:
This looks great. It's probably missing a header and shouldn't be console logging

mozartfish commented:
> This looks great. It's probably missing a header and shouldn't be console logging

removed print statements with this commit https://github.com/multinet-app/multimatrix/pull/226/commits/54f046197792e7a81a7c35710a295764a645a0b1

added child aggregation count with this commit https://github.com/multinet-app/multimatrix/pull/226/commits/e55951fef997f6d0664ba5f30b9aa985079a599e

JackWilb commented on src/components/MultiMatrix/MultiMatrix.vue:
```diff
@@ -1207,6 +1273,16 @@ export default Vue.extend({
           }
         });
 
+      // Add Children Count Label
+      attributeRowsEnter
+        .append('text')
+        .style('font-size', '10px')
+        .style('text-transform', 'capitalize')
+        .style('word-wrap', 'break-word')
+        .attr('text-anchor', 'left')
+        .attr('transform', 'translate(258, 0)')
+        .text('# children');
+
```
This should only happen if the matrix is aggregated and be removed when we're not aggregating

mozartfish replied:
fixed with this commit https://github.com/multinet-app/multimatrix/pull/226/commits/54463bfbab3f5e9135bfc70e485788a8910201a1

mozartfish replied:
fixed with this commit https://github.com/multinet-app/multimatrix/pull/226/commits/5542450b35e496554c1812ef183a38b4ac6c9441

JackWilb commented on src/components/MultiMatrix/MultiMatrix.vue:
```diff
@@ -1321,6 +1436,38 @@ export default Vue.extend({
+      const labelContainerWidth = rowLabelContainerStart;
+
+      // Draw Super Children Label Count
+      attributeVisEnter
+        .append('foreignObject')
+        .attr('x', () => {
+          return 270;
+        })
+        .attr('y', -5)
+        .attr('width', () => {
+          return labelContainerWidth - 50;
+        })
+        .attr('height', labelContainerHeight)
+        .classed('countForeign', true)
+        .append('xhtml:p')
+        .text((d: Node) => {
+          if (d.type === 'supernode') {
+            return d.CHILD_COUNT;
+          } else {
+            return '--';
+          }
+        })
+        .style('color', () => {
+          return 'black';
+        })
+        .classed('countLabels', true);
```
Same here, only if the matrix is aggregated

mozartfish replied:
fixed with this commit https://github.com/multinet-app/multimatrix/pull/226/commits/54463bfbab3f5e9135bfc70e485788a8910201a1

mozartfish replied:
fixed with this commit https://github.com/multinet-app/multimatrix/pull/226/commits/5542450b35e496554c1812ef183a38b4ac6c9441

JackWilb commented on src/components/MultiMatrix/MultiMatrix.vue:
```diff
@@ -1329,7 +1476,7 @@ export default Vue.extend({
         ),
       ];
-      return uniqueValues.length > 5;
+      return uniqueValues.length > uniqueValues.length + 1;
```
This is always false. It's basically:

```
return x > x+1
```
and x has is a non-negative integer. We should just switch this to 15 instead of 5.

mozartfish replied:
fixed with this commit https://github.com/multinet-app/multimatrix/pull/226/commits/3c8c4e7cb7ca3f006cf32b073269d0ec43e70f3d

JackWilb replied:
Added my desired fix in 63acc7e
